cypress/integration/main-page_spec.js

```diff
@@ -0,0 +1,63 @@
+describe('main page', () => {
+  beforeEach(() => {
+    cy.fixture('./stocks.json').then((allStocks) => {
+      cy.intercept('GET', 'https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=30&page=1&sparkline=false', {
+        statusCode: 200,
+        body: allStocks
+      })
+    cy.visit('http://localhost:3000/');
+    })
+  })
+
+  it('should contain a header', () => {
+    cy.contains('h1', 'MARKET WATCH')
+  })
+
+  it('should render stock cards upon page visit', () => {
+    cy.get('.stock')
+      .should('have.length', 7)
+  })
+
+  it('should render stock cards with stock names', () => {
+    cy.contains('Bitcoin').should('be.visible')
+    cy.contains('Ethereum').should('be.visible')
```

Comment:
One way to check these are rendering is cy.get(what your searching for).eq(0).contains('Bitcoin')
It replaces the visibility. What you have obviously works as well!